examples/bq-email-exports/terraform/variables.tf

```diff
@@ -33,32 +33,70 @@ variable "service_acct_roles" {
   description = "Roles for the Cloud Function service account"
   default     = ["roles/bigquery.admin", "roles/storage.admin", "roles/iam.serviceAccountTokenCreator"]
 }
-variable "function_name" {
-  description = "Name for the Cloud Function"
+
+variable "topic_name_1" {
+  description = "Name for the Pub/Sub topic that is triggered by Cloud Scheduler."
+  default     = "bq-email-run-query"
 }
 
-variable "topic_name" {
-  description = "Name for the Pub/Sub topic that the Cloud Function will subscribe to"
+variable "topic_name_2" {
+  description = "Name for the Pub/Sub topic that subscribes to the audit log sink for completed queries."
+  default     = "bq-email-export-gcs"
 }
-variable "sendgrid_api_key" {
-  description = "API key for authenticating the sending of emails through SendGrid API"
+
+variable "topic_name_3" {
+  description = "Name for the Pub/Sub topic that subscribes to the audit log sink for exported queries."
+  default     = "bq-email-send-email"
+}
+
+variable "query_logging_sink_name" {
+  description = "Name for the logging sink that triggers Pub/Sub topic 2 when query is completed."
+  default     = "bq-email-query-completed"
+}
+
+variable "export_logging_sink_name" {
+  description = "Name for the logging sink that triggers Pub/Sub topic 3 when export to GCS is completed."
+  default     = "bq-email-export-completed"
 }
 
-variable "bq_dataset" {
-  description = "Dataset name where BQ queries will run."
+variable "sendgrid_api_key" {
+  description = "API key for authenticating the sending of emails through SendGrid API"
 }
 
 variable "storage_bucket" {
-  description = "Storage bucket for exported JSON files."
+  description = "GCS bucket to store exported query results from BQ."
 }
 
 variable "bucket_lifecycle" {
-  description = "Number of days the exported JSON query result files should stay in the storage bucket."
-  default     = 7
+  description = "Number of days the exported query result files should stay in the storage bucket."
+  default     = 1
+}
+
+variable "function_bucket_1" {
```

Comment:
All 3 functions can probably just put their source in a single "cloud_functions_source_bucket". What's the value of a bucket for each function?

Reply:
This is because Terraform bundles all of the files inside of the provided GCS bucket into one Cloud Function. This way the respective `main.py` and `requirements.txt` file in each bucket can have a unique CF.